lib/rspec/core/shared_example_group.rb

```diff
@@ -213,16 +213,35 @@ def valid_name?(candidate)
 
         def warn_if_key_taken(context, key, new_block)
           existing_module = shared_example_groups[context][key]
-
           return unless existing_module
 
-          RSpec.warn_with <<-WARNING.gsub(/^ +\|/, ''), :call_site => nil
-            |WARNING: Shared example group '#{key}' has been previously defined at:
-            |  #{formatted_location existing_module.definition}
-            |...and you are now defining it at:
-            |  #{formatted_location new_block}
-            |The new definition will overwrite the original one.
-          WARNING
+          old_definition = formatted_location existing_module.definition
+          new_definition = formatted_location new_block
+          loaded_spec_files = RSpec.configuration.loaded_spec_files
+
+          if loaded_spec_files.include?(new_definition) && old_definition == new_definition
```

Comment:
> I think this warning should be displayed for any examples defined in spec locations, it'll be more useful.

I disagree.  I think the improved phrasing being added here would be inaccurate and confusing in cases the locations are not identical.  If the locations are not identical, then the user really is defining two groups with the same name, and the original warning (which prints the two different locations and says the new one will overwrite the original one) is accurate and clear.

Consider also that putting a shared example group in a file ending in `_spec.rb` isn't an error or even something to warn about in general.  Maybe the user is defining a scoped shared group within an example group, and accidentally copied and pasted it so that the same group was defined twice in the same spec file.  Suggesting they rename their file would be wrong in that case.  Or maybe they want to name their support files so that RSpec loads them automatically and they don't have to require them manually.  I wouldn't follow that approach myself but there's nothing wrong with it per-se.